app/services/TaxCreditsService.scala

```diff
@@ -30,12 +30,14 @@ class TaxCreditsService @Inject() (taxCreditsConnector: TaxCreditsConnector)(imp
   def checkForTaxCredits(nino: Option[Nino])(implicit headerCarrier: HeaderCarrier): Future[Option[Boolean]] =
     if (nino.isEmpty) {
       Future.successful(Some(false))
-    } else {
-      taxCreditsConnector.checkForTaxCredits(nino.get) bimap (
-        error => if (error.statusCode == NOT_FOUND) Some(false) else None,
-        result => {
-          if (result.status == OK) Some(true) else Some(false)
-        }
-      )
-    }.merge
+    } else
+      taxCreditsConnector
+        .checkForTaxCredits(nino.get)
+        .bimap(
```

Comment:
Can you fold here instead of bimap to remove the `.merge`

Reply:
Done